### PR TITLE
Feat: liquid staking module sketch #LSC-267

### PR DESCRIPTION
**Pull request by zavgorodnii**

### CODE REVIEW

**Comment by zavgorodnii** [comment]
a quick note (just thought about it) — what if we receive a big ass voucher that will mess up our distribution? we might want to consider redelegating or something.

File: contracts/lido_cosmos_hub/src/testing/tests.rs [hunk 1]
```diff
@@ -129,6 +132,31 @@ pub fn do_register_validator(
...
 }
 
+pub fn do_register_tokenize_share_record(
+    deps: &mut OwnedDeps<MockStorage, MockApi, WasmMockQuerier>,
+    record: TokenizeShareRecord,
+) {
+    deps.querier.add_tokenize_share_record(record);
+}
+
+fn new_tokenize_share(
```
Comment (foxpy) [inline contracts/lido_cosmos_hub/src/testing/tests.rs]: Shouldn't this function be a static method of TokenizeShareRecord?

**Review by foxpy — CHANGES_REQUESTED** [review]

File: contracts/lido_cosmos_hub/src/tokenized.rs [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    deps: Deps,
+    validators_registry_contract: Addr,
+) -> StdResult<ValidatorResponse> {
+    deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+        contract_addr: validators_registry_contract.to_string(),
+        msg: to_binary(&QueryValidators::GetLargestValidator {})?,
+    }))
+}
+
+// TODO: remove unwraps
```
Comment (zavgorodnii) [inline contracts/lido_cosmos_hub/src/tokenized.rs]: todo comment

File: contracts/lido_cosmos_hub/src/tokenized.rs [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        .statom_token_contract
+        .ok_or_else(|| StdError::generic_err("the token contract must have been registered"))?;
+
+    let mut messages: Vec<CosmosMsg> = vec![];
+    let mut attrs: Vec<Attribute> = vec![];
+
+    // Trying to get TokenizeShareRecord for every voucher we've got by denom.
+    // If we can't find a record or we get an error, tx reverted
+    //
+    // Note: tokenized share denom looks like this:
```
Comment (zavgorodnii) [inline contracts/lido_cosmos_hub/src/tokenized.rs]: not sure this comment is necessary anymore

**Comment by zavgorodnii** [comment]
what's up with the idea to get the redeemed amount inside the `reply` handler?

**Comment by pr0n00gler** [comment]
> what's up with the idea to get the redeemed amount inside the `reply` handler?

I suggest to stick with the current version for now. When we will be able to test the contracts on more realistic infrastructure (our integration-test framework), we'll compare these two versions (current and with the 'reply'-handler) and choose a more accurate one (or with less gas consumption).

File: packages/basset/src/hub.rs [hunk 4]
```diff
@@ -135,45 +132,9 @@ pub enum Cw20HookMsg {
...
-    pub statom_applied_exchange_rate: Decimal,
-    pub statom_withdraw_rate: Decimal,
-
-    pub released: bool,
-}
-
-#[derive(JsonSchema, Serialize, Deserialize, Default)]
-pub struct UnbondWaitEntity {
-    pub statom_amount: Uint128,
+    pub max_burn_ratio: Decimal,
```
Comment (foxpy) [inline packages/basset/src/hub.rs:137]: We use both `max_burn_rate` and `max_burn_ratio` throughout code base. This is confusing.

**Review by foxpy — APPROVED** [review]

**Pull request merged**